lumicks/pylake/tests/test_imaging_confocal/test_kymo_from_corrstack.py

```diff
@@ -0,0 +1,164 @@
+import pytest
+import numpy as np
+from lumicks.pylake.correlated_stack import CorrelatedStack
+from lumicks.pylake.detail.widefield import TiffStack, Tether
+from lumicks.pylake.tests.data.mock_widefield import MockTiffFile
+from lumicks.pylake.kymo import _kymo_from_correlated_stack
+
+
+def make_frame_times(n_frames, rate=10, step=8, start=10, framerate_jitter=0, step_jitter=0):
+    def noise(jitter, j):
+        if j / rate % 2:
+            return 0
+        else:
+            return jitter
+
+    return [
+        [
+            f"{j + noise(framerate_jitter, j)}",
+            f"{j + step + noise(framerate_jitter, j) + noise(step_jitter, j)}",
+        ]
+        for j in range(start, start + n_frames * rate, rate)
+    ]
+
+
+def create_mock_corrstack(
+    n_frames=1, shape=None, image=None, framerate_jitter=0, step_jitter=0, tether_ends=None
+):
+    image = np.ones(shape) if image is None else image
+    times = make_frame_times(n_frames, framerate_jitter=framerate_jitter, step_jitter=step_jitter)
+    tether = tether_ends if tether_ends is None else Tether((0, 0), tether_ends)
+    fake_tiff = TiffStack(
+        [MockTiffFile(data=[image] * n_frames, times=times)],
+        align_requested=False,
+        tether=tether,
+    )
+    return CorrelatedStack.from_dataset(fake_tiff)
+
+
+def gaussian_2d(shape=(15, 15), limit=(3.0, 3.0), mean=(0.0, 0.0), std=(1.0, 1.0), rho=0.0):
+    """Create a 2D gaussian with two different standard deviations
+
+    Parameters
+    ----------
+    shape : tuple of int
+        The shape of the resulting 2D array for x and y.
+    limit : tuple of float
+        The absolute values of standard deviation up to where the values for the gaussians in x and
+        y are calculated.
+    mean : tuple of float
+        The mean of the guassians in values of standard deviation in x and y. 0.0 corresponds to the
+        image center of the corresponding axis.
+    std : tuple of float
+        The standard deviation of the two gaussians in x and y.
+    rho : float
+        The correlation of the two gaussians. Accepted values are in the open interval ( -1 , 1 ).
+    """
+    nx, ny = shape
+    dx, dy = limit
+    mx, my = mean
+    sx, sy = std
+
+    x, y = np.meshgrid(np.linspace(-dx, dx, nx), np.linspace(-dy, dy, ny))
+
+    def gauss2d(x=0.0, y=0.0, mx=0.0, my=0.0, sx=1.0, sy=1.0, rho=0.0):
+        return np.exp(
+            -(
+                ((x - mx) / sx) ** 2
+                - 2.0 * rho * ((x - mx) / sx) * ((y - my) / sy)
+                + ((y - my) / sy) ** 2
+            )
+            / (2.0 * (1.0 - rho**2))
+        ) / (2.0 * np.pi * sx * sy * np.sqrt(1.0 - rho**2))
+
+    return gauss2d(x, y, mx, my, sx=sx, sy=sy, rho=rho)
```

Comment:
I'd kind of prefer if we just use scipy for this.
```suggestion
    from scipy.stats import multivariate_normal

    sx, sy = std
    cov = np.array([[sx**2, sx * sy * rho], [sx * sy * rho, sy**2]])
    mesh = np.dstack(np.mgrid[[slice(-dx, dx, n * 1j) for (dx, n) in zip(limit, shape)]])
    return multivariate_normal(mean=mean, cov=cov).pdf(mesh)
```